### DISCUSSION TITLE
Welcome to irmin Discussions!

### DISCUSSION THREAD

**Discussion by samoht** [mirage/irmin#1185] [General]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by kluvin** [comment]
Hi all! I am working on a markdown reimplementation of WikiWikiWeb, where I am now extending this to similar problems as the federated wiki -- which is essentially taking git semantics and applying them to a wiki. The idea was originally to use git directly, although, since it lives only in the filesystem, that brings some problems; so I looked at several technologies that are labelled as git-for-x, irmin stood out in that it could directly operate on `.git` repositories.

I am curious about how the storage-agnostic nature of irmin, together with the git-backend compares to libgit2, which is used in GitHub and others to programmatically manage bare repos. In fact libgit2 operates on disk as much as git itself does, and so I figure there is significant performance gains to be achieved by operating in memory, and only flushing to disk sometimes. Can irmin do something like that? Please let me know if anything is not clear.

PS: I am getting something like this:

```shell
$ irmin set x 1
$ irmin get x
> irmin: [ERROR] The client wants to refill the internal buffer by a bigger input.
  ERROR: Chunk loaded from ./.git/objects/be/eac1f72988177236341222008601144f5950da is too big (65536 bytes) for the decoder (max: 65536 bytes)
$ git gc
> ...done
$ irmin get x
> 1
```

When getting an object after writing a commit with `irmin set`, a simple `git gc` fixes this until the next write. It seems similar to #1001. Shall I publish an issue with more details?

**Comment by Ngoguey42** [comment]
Hi @kluvin. Can you open a dedicated topic? You have answered to the welcome message.
